pylint/testutils/output_line.py

```diff
@@ -62,39 +76,42 @@ def __init__(self, row, exception):
 class OutputLine(NamedTuple):
     symbol: str
     lineno: int
-    column: str
-    object: Any
+    column: int
+    object: str
     msg: str
-    confidence: interfaces.Confidence
+    confidence: str
 
     @classmethod
-    def from_msg(cls, msg):
+    def from_msg(cls, msg: Message) -> "OutputLine":
         column = cls.get_column(msg.column)
         return cls(
             msg.symbol,
             msg.line,
             column,
             msg.obj or "",
             msg.msg.replace("\r\n", "\n"),
-            msg.confidence.name
-            if msg.confidence != interfaces.UNDEFINED
-            else interfaces.HIGH.name,
+            msg.confidence.name if msg.confidence != UNDEFINED else HIGH.name,
         )
 
     @classmethod
-    def get_column(cls, column):
-        if not PY38_PLUS:
-            return ""  # pragma: no cover
-        return str(column)
+    def get_column(cls, column: str) -> int:
+        if (
+            not PY38_PLUS
+        ):  # This mimicks behaviour of MessagesHandlerMixIn.add_one_message()
+            return 0  # pragma: no cover
+        return int(column)
 
     @classmethod
-    def from_csv(cls, row):
+    def from_csv(cls, row: Union[Tuple[str, ...], List[str], str]) -> "OutputLine":
         try:
-            confidence = row[5] if len(row) == 6 else interfaces.HIGH.name
             column = cls.get_column(row[2])
-            return cls(row[0], int(row[1]), column, row[3], row[4], confidence)
+            if isinstance(row, Iterable) and len(row) == 6:
+                return cls(row[0], int(row[1]), column, row[3], row[4], row[5])
+            if isinstance(row, Iterable) and len(row) == 5:
+                return cls(row[0], int(row[1]), column, row[3], row[4], HIGH.name)
+            raise IndexError
```

Comment:
Maybe we could uniformize the expected text output by adding the confidence everywhere, so there's only one type of output and we can treat all expected message the same ? (This would prevent big diff where we add the confidence after upgrading the expected output). @cdce8p what do you think, I remember that we had this conversation already at some point.

Reply:
This should definitely be part of a separate PR and would then ideally follow after this. If desired I can go and create this. Should be as easy as running `python tests/test_functional.py --update-functional-output` and updating the typing and this function here.

Reply:
Just my personal opinion: I don't see much value in adding `HIGH` to almost every pylint error in all functional tests. Especially since it's the last value, it would be much more useful to only specify it if it's different (i.e. not `HIGH`).

Reply:
While I agree with the no added value argument, I think Pierre makes a valid point that when we do this once we will then have less larger diffs after people use the `--update-functional-output` to update tests after they have added new cases. I saw this during my own work as well, where all test data gets updated because of it.
I don't think it would hurt to do this once and would allow simplifying the code here.

Reply:
We can update / revert `--update-functional-output` to only add different confidence values 😜 Then we would only need to update a few.

Anyway, doesn't matter the decision, this should be part of another PR. (I saw a question about that earlier.)